node/builder.go

```diff
@@ -376,14 +376,21 @@ func (b *Builder) buildNetwork(ctx context.Context, config *config.BootstrapConf
 }
 
 func (b *Builder) buildMessaging(ctx context.Context, network *NetworkSubmodule, chain *ChainSubmodule, wallet *WalletSubmodule) (MessagingSubmodule, error) {
-	msgPool := message.NewPool(b.Repo.Config().Mpool, consensus.NewIngestionValidator(chain.State, b.Repo.Config().Mpool))
+	poolValidator := consensus.NewIngestionValidator(chain.State, b.Repo.Config().Mpool)
+	msgPool := message.NewPool(b.Repo.Config().Mpool, poolValidator)
```

Comment:
Does this mean we're doing validation twice, once in the pubsub and then again in the pool? We shouldn't do that, but have clear expectations about what is validated when. 

Validating syntax at the network makes sense, but I'm much less confident that the more expensive validation of comparing the message with blockchain state does. Maybe the spec people have some insight there?

Reply:
> Does this mean we're doing validation twice

Yup

> We shouldn't do that, but have clear expectations about what is validated when

I agree, this is one of those imperfect steps towards untangling the massive knot of what-gets-validated-when.  There is no reasonable single PR that is going to get us where we want to go.  However it is noted that I have under-communicated this objective and I will create an issue documenting thought process and evolving plan.

> Maybe the spec people have some insight there?

We can wait for more details but right now the spec claims " the account in question must have enough funds to cover the actions specified" in order for propagation validation to pass.  This is enough for me to want to plug full validation into this part of the system as a first pass.